tests/profitbricks/requests/compute/server_tests.rb

```diff
@@ -228,20 +285,39 @@
       detachVolumeResponse.status == 202
     end
 
-    tests('#attach_cdrom').data_matches_schema(@resource_schema) do
-      attachCdromResponse = service.attach_cdrom(@datacenter_id, @server_id, @image_id)
+    if ENV["FOG_MOCK"] == "false"
+      tests('#attach_cdrom').data_matches_schema(@resource_schema) do
+        attachCdromResponse = service.attach_cdrom(@datacenter_id, @server_id, @image_id)
 
-      @cdrom_id = attachCdromResponse.body['id']
+        @cdrom_id = attachCdromResponse.body['id']
 
-      attachCdromResponse.body
+        attachCdromResponse.body
+      end
     end
 
-    tests('#get_attached_cdrom').data_matches_schema(@resource_schema) do
-      if ENV["FOG_MOCK"] != "true"
+    if ENV["FOG_MOCK"] == "true"
+      tests('#attach_cdrom').succeeds do
+        attachCdromResponse = service.attach_cdrom(@datacenter_id, @server_id, @image_id)
+
+        @cdrom_id = attachCdromResponse.body['id']
+
+        attachCdromResponse.body
+      end
+    end
+
+    if ENV["FOG_MOCK"] == "false"
+      tests('#get_attached_cdrom').data_matches_schema(@resource_schema) do
         sleep(60)
+        getAttachedVolumeResponse = service.get_attached_cdrom(@datacenter_id, @server_id, @cdrom_id)
+        getAttachedVolumeResponse.body
+      end
+    end
+
+    if ENV["FOG_MOCK"] == "true"
```

Comment:
You should be able to use [`Fog.mock?`](https://github.com/fog/fog-core/blob/188c1253bd51a9b1aeba7e3ab675de8e677aaada/lib/fog/core/mock.rb#L15) here.